source/github.php

```diff
@@ -0,0 +1,57 @@
+ <?php
+
+// Check for local file existence, set ETag value to true if file doesn't exist
+// to force curl of API
+if ($github_file = file_get_contents('github_api_output.json')) {
+    $github_json = json_decode($github_file, true);
+} else {
+    $github_json = ['headers' => ['Last-Modified' => false]];
+}
+
+// Send API request to GitHub, including authentication information to ensure rate limits are higher
+// See https://developer.github.com/v3/#increasing-the-unauthenticated-rate-limit-for-oauth-applications for details
+$curl = curl_init();
+$url = 'https://api.github.com/repos/openshift/origin/releases/latest';
+$data = ['client_id' => '7e65d957b9bb631e53e6',
+         'client_secret' => '6ce97999e104ed75fded7328485a5f0e36357a46'];
```

Comment:
This [should not be shared](https://developer.github.com/guides/basics-of-authentication/#registering-your-app) publicly in the repo.